qiskit/utils/deprecation.py

```diff
@@ -67,3 +78,23 @@ def _rename_kwargs(func_name, kwargs, kwarg_map):
             )
 
             kwargs[new_arg] = kwargs.pop(old_arg)
+
+
+def _extend_docstring(func, msg_lines):
+    docstr = func.__doc__
+    if docstr:
+        docstr_lines = docstr.expandtabs().splitlines()
+    else:
+        docstr_lines = ["DEPRECATED"]
+    indent = 1000
+    for line in docstr_lines[1:]:
+        stripped = line.lstrip()
+        if stripped:
+            indent = min(indent, len(line) - len(stripped))
+    spaces = ""
+    if indent != 1000:
+        spaces = " " * indent
+    new_doc_str_lines = [docstr_lines[0], spaces + ".. warning::"]
+    for msg_line in msg_lines:
+        new_doc_str_lines.append(spaces + msg_line)
```

Comment:
You may also be interested in swapping to using the
```rst
.. deprecated:: <version>
  <message>
```
role in reST instead - it's a bit more specific than a warning.

Reply:
I'm learning a lot.. thanks! :)

Reply:
The theme does not have support for it. If https://github.com/Qiskit/qiskit_sphinx_theme/issues/71 is a quick fix, we can block this PR on it. Otherwise, what do you think about `warning`?